client/question/services/ConversationManagerService.js

```diff
@@ -13,18 +13,48 @@
 // limitations under the License.
 
 /**
- * @fileoverview Service that orchestrates the processing of a user's code
- * submission.
+ * @fileoverview Service that orchestrates TIE's conversation with the learner.
  */
 
-tie.factory('SolutionHandlerService', [
+tie.factory('ConversationManagerService', [
   '$q', 'CodePreprocessorDispatcherService', 'CodeRunnerDispatcherService',
   'FeedbackGeneratorService', 'PrereqCheckDispatcherService',
-  'TranscriptService', 'CodeSubmissionObjectFactory',
+  'TranscriptService', 'CodeSubmissionObjectFactory', 'FEEDBACK_CATEGORIES',
+  'FeedbackDetailsObjectFactory',
   function(
       $q, CodePreprocessorDispatcherService, CodeRunnerDispatcherService,
       FeedbackGeneratorService, PrereqCheckDispatcherService,
-      TranscriptService, CodeSubmissionObjectFactory) {
+      TranscriptService, CodeSubmissionObjectFactory, FEEDBACK_CATEGORIES,
+      FeedbackDetailsObjectFactory) {
+
+    // TODO(sll): Modify this to also depend on the learner's submission
+    // history.
```

Comment:
Depend on a history how? Sorry, I'm gonna stickler on the TODOs here just in case they end up dangling.

Reply:
Thanks! Clarified, PTAL.